replica/table.cc

```diff
@@ -2937,13 +2951,23 @@ table::query(schema_ptr s,
         co_return make_lw_shared<query::result>();
     }
 
-    _async_gate.enter();
+    const auto table_async_gate_holder = _async_gate.hold();
+    utils::small_vector<gate::holder, 1> cg_async_gate_holders;
+    {
+        auto it = partition_ranges.begin();
+        cg_async_gate_holders = hold_gate_for_compaction_groups(*it++);
```

Comment:
nit: iterator post-increment is typically less efficient than pre-increment since it needs to copy and return `*this`.
How about the following?
```c++
        cg_async_gate_holders = hold_gate_for_compaction_groups(*it);
        // in most cases partition_ranges.size() == 1, so the below inefficient loop is no-op.
        while (++it != partition_ranges.end()) {
            auto holders = hold_gate_for_compaction_groups(*it);
            std::ranges::move(holders, std::back_inserter(cg_async_gate_holders));
        }
```

Reply:
Done in v6.